test/ts-scenario/steps/lib/gateway.ts

```diff
@@ -308,55 +306,56 @@ export async function performHandledGatewayTransaction(gatewayName: string, ccNa
 
 	// Build a transaction
 	const transaction: Transaction = contract.createTransaction(func);
-	const transactionId: string = transaction.getTransactionID().getTransactionID();
 
 	// Submit/evaluate transaction
 	if (submit) {
-		// Create event hubs to monitor so we can test it worked
-		const org1EventHub: ChannelEventHub = await getInternalEventHubForOrg(gateway, channelName, 'Org1MSP');
-		const org2EventHub: ChannelEventHub = await getInternalEventHubForOrg(gateway, channelName, 'Org2MSP');
-
-		let org1EventsFired: any = 0;
-		let org2EventsFired: number = 0;
-		org1EventHub.registerTxEvent('all', (txId: string, code: string) => {
-			if (code === 'VALID' && txId === transactionId) {
-				org1EventsFired++;
-			}
-		}, () => {
-			// Ignore errors
-		});
-		org2EventHub.registerTxEvent('all', (txId: string, code: string) => {
-			if (code === 'VALID' && txId === transactionId) {
-				org2EventsFired++;
-			}
-		}, () => {
-			// Ignore errors
+		let listener: CommitEventListener;
+		let resolveNotificationPromise: () => void;
+		let rejectNotificationPromise: () => void;
+		let eventResults: any;
+
+		const notificationPromise = new Promise((resolve, reject) => {
+			resolveNotificationPromise = resolve;
+			rejectNotificationPromise = reject;
 		});
 
+		listener = await network.addCommitListener(
+			async (error: Error, blockNum: string, txid: string, status: string) => {
+				if (error) {
+					rejectNotificationPromise();
+				} else {
+					eventResults = {txid, status, blockNum};
+					resolveNotificationPromise();
+				}
+			},
+			{} // options
+		);
+
 		try {
-			// Split args, do not capture response
+			// -------- S E N D
 			await transaction.submit(...funcArgs);
 			BaseUtils.logMsg(`Successfully submitted transaction [${func}] using handler [${EventStrategies[handlerOption]}]`);
+
+			await notificationPromise;
+			listener.unregister();
+			BaseUtils.logMsg(`Successfully got event status [${func}] using handler [${EventStrategies[handlerOption]}]`, undefined);
+			gatewayObj.result = {type: 'event', response: JSON.stringify(eventResults)};
```

Comment:
Does changing the code flow so results are only set in a non-error condition mean that subsequent checks of the results may be checking old results instead of current ones? This looks like a potential test bug